CoreObjectModel/SourceModel/Interfaces.cs

```diff
@@ -1540,6 +1540,31 @@ abstract class ISourceLocationProviderContract : ISourceLocationProvider {
   }
   #endregion
 
+  /// <summary>
+  /// An object that can retrieve the source link information embedded in a PDB file.
+  /// For example, a PDB reader that can read the source link information
+  /// </summary>
+  [ContractClass(typeof(ISourceLinkProviderContract))]
+  public interface ISourceLinkProvider {
+    /// <summary>
+    /// Return the source link data as a byte array or null if there is no source link information embedded
+    /// </summary>
+    IEnumerable<byte> GetSourceLinkData();
```

Comment:
I'd just return `byte[]`, like Checksum property declared also in this file.

Reply:
@MichalStrehovsky

Reply:
The CCI convention is to return `IEnumerable<byte>` for these. The Checksum property was added by a Project N developer and doesn't follow CCI conventions. It's not an example to follow.

Reply:
Is there another example of API that returns `IEnumerable<byte>`?

Reply:
> Is there another example of API that returns IEnumerable<byte>?

There's many. `IAssemblyReference.HashValue`. `IAssemblyReference.PublicKey`. `IAssemblyReference.PublicKeyToken`. `IResource.Data`.

I agree that most of them are frustrating to deal with because of that. Sometimes I think "frustrating to deal with" is the point of CCI.

Reply:
I see. Well, the first thing that one would want to do with the SourceLink data is to convert to a string using `Encoding.UTF8.GetString` and that doesn't take `IEnumerable<byte>`.